disk/src/com/treode/disk/edit/PageReader.scala

```diff
@@ -27,16 +29,33 @@ private class PageReader (
   val bits = 10
   val buffer = PagedBuffer (bits)
 
+  /*
+      We read a P-type object from the location at pos using the 
+      Pickler embedded in the PageDescriptor to unpickle the object  
+   */
+  def read [P] (pd : PageDescriptor[P], pos: Position): Async[P] = {
+    buffer.clear()
+    assert(pos.disk == 0)
+    for {
+      _ <- file.fill (buffer, pos.offset, pos.length) 
+    } yield {
+      pd.ppag.unpickle(buffer)
+    }
+  }
+
+
   /**
    * Returns (if successful) the string of length `length` at `pos` in the
    * file asynchronously, using a read buffer.
    */
-  def readString (pos: Long, length: Int) : Async [String] = {
+  def readString (pos: Long, length: Long) : Async [String] = {
```

Comment:
I think you should not need this anymore. I see you are using it in the spec, but you can use the above in the spec by giving it a pickler for a string. Remove it and change the spec.